src/dmd/initsem.d

```diff
@@ -604,6 +610,19 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, ref Typ
             return err();
         }
 
+        /* If it's an array of integral being initialized by `{ string }`
+         * replace with `string`
+         */
+        auto tn = t.nextOf();
+        if (tn.isintegral())
+        {
+            if (ExpInitializer ei = isBraceExpression())
```

Comment:
maybe `auto ei = ...`?

Reply:
I would use `auto` if it was obvious what type it would be, but it isn't obvious.